apps/frontend/src/components/TimeCardPage/TimeSheet.tsx

```diff
@@ -1,341 +1,415 @@
+          setSelectedUser(users[0]);
         });
-        setUserTimesheets(modifiedTimesheets);
-
-        //Also need to update our list of currently selected - TODO come up with a way to not need these duplicated lists
-        setCurrentTimesheets(currentTimesheets.map(
-            (entry) => {
-                if (entry.TimesheetID === selectedTimesheet.TimesheetID) {
-                    return {
-                        ...entry,
-                        TableData: updated_sheet.TableData
-                    }
-                }
-                return entry
-            }
-        ));
+      }
+      setSelectedUser(userInfo);
+    });
+    // if employee setSelectedUSer to be userinfo
+    // if supervisor/admin get all users
+    // set selected user
+  }, []);
+
+  const getUpdatedTimesheet = (userId) => {
+    apiClient.getUserTimesheets(userId).then((timesheets) => {
+      setUserTimesheets(timesheets);
+      //By Default just render / select the first timesheet for now
+      setCurrentTimesheetsToDisplay(timesheets, selectedDate);
+    });
+  };
+
+  // Pulls user timesheets, marking first returned as the active one
+  useEffect(() => {
+    getUpdatedTimesheet(selectedUser?.UserID);
+  }, [selectedUser]);
+
+  // Callback function that triggers GET request to the API to grab the most recent version of timesheets for the current selected user,
+  // and re-sets the current timesheet state variable
+  const forceRefreshTimesheet = () => {
+    if (selectedUser !== undefined) {
+      getUpdatedTimesheet(selectedUser.UserID);
+    } else {
+      console.log("ERROR: Couldn't force refresh timesheet");
     }
-
-    const updateDateRange = (date: Moment) => {
-        setSelectedDate(date);
-        //TODO - Refactor this to use the constant in merge with contants branch
-        setCurrentTimesheetsToDisplay(userTimesheets, date);
-    }
-
-    const changeTimesheet = (sheet) => {
-        setTimesheet(sheet)
-        setWeeklyComments(getAllActiveCommentsOfType(CommentType.Comment, sheet.WeekNotes))
-        setWeeklyReports(getAllActiveCommentsOfType(CommentType.Report, sheet.WeekNotes))
-    }
-
-
-    const setCurrentTimesheetsToDisplay = (timesheets, currentStartDate: Moment) => {
-        const newCurrentTimesheets = timesheets.filter(sheet => moment.unix(sheet.StartDate).isSame(currentStartDate, 'day'));
-
-        setCurrentTimesheets(newCurrentTimesheets);
-        if (newCurrentTimesheets.length > 0) {
-            changeTimesheet(newCurrentTimesheets[0])
+  };
+
+  const processTimesheetChange = (updated_sheet) => {
+    // Updating the rows of the selected timesheets from our list of timesheets
+    const modifiedTimesheets = userTimesheets.map((entry) => {
+      if (entry.TimesheetID === selectedTimesheet.TimesheetID) {
+        return {
+          ...entry,
+          TableData: updated_sheet.TableData,
+        };
+      }
+      return entry;
+    });
+    setUserTimesheets(modifiedTimesheets);
+
```

Comment:
has anything changed in this file or is it all prettier?

Reply:
Yes, just tagged you in the places where actual code changes happened